_posts/2021-07-13-gs_00003_terminal_basics2.md

```diff
@@ -0,0 +1,394 @@
+list. An often used command is
+
+    $ ps aux
+
+which gives a list of all running processes. Since that list is usually long,
+you can search for a specific process:
+
+    $ ps aux | grep firefox
+
+You should be able to find your PID and also might see that firefox created
+process children. If you kill the main process, you will kill all children as
+well. Damn ... This world is so cruel! Sometimes it is also nice to use:
+
+    $ ps axjf
+
+This will show you a process tree. You can see that firefox is a child of
+something called "bash". We will talk about it in the next post, but it means
+your terminal. Again, you can see the children of firefox.
+
+One last thing! Sometimes, it is useful to monitor the CPU utilization and see
+which process uses it. A nice tool to do so is:
+
+    $ htop
+
+Our computer uses right now 814 MB of RAM and has a CPU utilization of around
+20%. What about yours? If you have multiple cores, you can see the utilization
+per core. Exit "htop" using CTRL-C.
+
+<div class='note-example' markdown="1">
+Real-life Example:
+
+You notice that your computer is suddenly very slow. Especially in software
+development, a process might use a huge amount of computing power due to a bug.
+This can have a slowdown of your computer as consequence. There could also be a
+process in the background that did not close as expected. With "htop", you can
+identify such processes and kill them.
+
+</div>
+
+# Summary
+
+In the last two posts you have learned in total 17 commands:
+
+    pwd, ls, cd, touch, mkdir, cp, mv, rm, cat, nano, find, grep, pacman, kill, killall, ps, htop
+
+If you compare that with the number of vocabularies you have to learn for a
+foreign language, it is really not much. To be fair, there are many parameters
+you can use with the commands, but still, it should just take a little practice
+to learn them. And if you forgot something, there is still the man page where
+you can look it up. Of course, there are dozens of other commands. But with
+these, you come already pretty far!
+
+Like the last post, we summarized the commands in a cheat sheet that you can
+find in the [GitHub repository](https://github.com/bitsandvolts/getting_started)
+of the Getting Started category. We put a lot of work and energy into our blog
+posts and the additional material, so please respect our
+[terms of usage](https://github.com/bitsandvolts/getting_started/blob/master/LICENSE).
+Besides the commands you've also learned:
+
+- Nano is one possible editor
+- Man page is a good place to look up commands
+- A package manager handles software-related things for you
+- Software can be dependent on other software
+- It is recommended to run a system update from time to time
+- The Arch Wiki is also a good resource
+- You need root privileges to install or deinstall system software
+- Programs can run as a background process
+- Process IDs (PIDs) uniquely identify a process
```

Comment:
```suggestion
- **Process IDs** (PIDs) uniquely identify a process
```

I think you didn't mention the uniqueness property of PIDs in the sections earlier.

Reply:
to be changed